src/DateTimeInput/index.jsx

```diff
@@ -151,221 +159,255 @@ function setPrecision( mode )
+
+        if ( /^\d\d?$/.test( trimmed ) && digits >= 0 && digits <= 59 )
+        {
+            const newTimestamp = $m( timestamp ).set( 'minute', digits )
+                .valueOf();
+
+            setTimestamp( newTimestamp );
+            setEditingMainInputValue( formatDateTime(
+                newTimestamp,
+                props.format || setPrecision( props.mode ),
+            ) );
+        }
+        else
+        {
+            digits = _.isNumber( timestamp ) && $m( timestamp ).minute();
+
+            if ( !_.isNaN( digits ) )
+            {
+                const newTimestamp = $m( timestamp ).set( 'minute', digits )
+                    .valueOf();
+
+                setTimestamp( newTimestamp );
+                setEditingMainInputValue( formatDateTime(
+                    newTimestamp,
+                    props.format || setPrecision( props.mode ),
+                ) );
+            }
+        }
+    }, [ props.format, props.mode, timestamp ] );
+
+
+    const canGotoNext = useCallback( () =>
     {
-        const { max } = this.props;
-        const nextGridStart = $m( this.state.gridStartTimestamp )
-            .add( 1, this.props.mode === 'month' ? 'year' : 'month' ).valueOf();
+        const { max } = props;
+        const nextGridStart = $m( gridStartTimestamp )
+            .add( 1, props.mode === 'month' ? 'year' : 'month' ).valueOf();
+
+        return !_.isNumber( max ) || ( nextGridStart <= max );
+    }, [ gridStartTimestamp, props.mode, props.max ] );
 
-        return !_.isNumber( max ) ||
-            ( nextGridStart <= max );
-    }
 
-    canGotoPrev()
+    const canGotoPrev = useCallback( () =>
     {
-        const min = this.props.min || now();
-        const prevGridStart = $m( this.state.gridStartTimestamp )
-            .add( -1, this.props.mode === 'month' ? 'year' : 'month' )
+        const min = props.min || now();
+        const prevGridStart = $m( gridStartTimestamp )
+            .add( -1, props.mode === 'month' ? 'year' : 'month' )
             .valueOf();
         const endOfPrev = $m( prevGridStart )
-            .add( 1, this.props.mode === 'month' ? 'year' : 'month' )
+            .add( 1, props.mode === 'month' ? 'year' : 'month' )
             .valueOf();
 
-        return !_.isNumber( min ) ||
-            endOfPrev > min;
-    }
+        return !_.isNumber( min ) || endOfPrev > min;
+    }, [ gridStartTimestamp, props.mode, props.min ] );
 
-    canEditHourOrMinute()
-    {
-        return _.isNumber( this.state.timestamp );
-    }
 
-    isUnitSelectable( timestamp, unit, allowFraction )
+    const canEditHourOrMinute = useCallback( () =>
+        _.isNumber( timestamp ), [ timestamp ] );
+
+
+    const isUnitSelectable = useCallback( (
+        newTimestamp = timestamp,
+        unit,
+        allowFraction,
+    ) =>
     {
-        const { max } = this.props;
-        const min = this.props.min || now();
+        const { max } = props;
+        const min = props.min || now();
```

Comment:
Using `now()` inside of `useCallback` is potentially problematic. I guess this should be extracted to outside of the DateTimeInput function

Reply:
Same applies to `canGotoPrev` which also uses `now()`

Reply:
As discussed in person, this won't actually be a problem, since we're only calling `now()` once if there's no `props.min`

Reply:
Indeed, my above comment was nonsense